core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -377,6 +382,23 @@ class KafkaServer(val config: KafkaConfig, time: Time = Time.SYSTEM, threadNameP
     clusterResourceListeners.onUpdate(new ClusterResource(clusterId))
   }
 
+  private[server] def notifyMetricsReporters(metricsReporters: Seq[AnyRef]): Unit = {
+    val metricsContext = createKafkaMetricsContext()
+    metricsReporters.foreach {
+      case x: MetricsReporter => x.contextChange(metricsContext)
+      case _ => //do nothing
+    }
+  }
+
+  private[server] def createKafkaMetricsContext() : KafkaMetricsContext = {
+    val metadata = new util.HashMap[String, Object]
+    metadata.put(KAFKA_CLUSTER_ID, clusterId)
+    metadata.put(KAFKA_BROKER_ID, config.brokerId.toString)
+    metadata.putAll(config.originalsWithPrefix(CommonClientConfigs.METRICS_CONTEXT_PREFIX))
```

Comment:
in the KIP we don't add metrics.context prefix values for kafka brokers

Reply:
config.originalsWithPrefix(prefix) is the same as config.originalsWithPrefix(prefix, true)
https://github.com/apache/kafka/blob/b72ab51e4d7cfe39094985d7a732618ef57b877f/clients/src/main/java/org/apache/kafka/common/config/AbstractConfig.java#L247

If strip set to true, it will remove the prefix:
https://github.com/apache/kafka/blob/b72ab51e4d7cfe39094985d7a732618ef57b877f/clients/src/main/java/org/apache/kafka/common/config/AbstractConfig.java#L261

Reply:
I have added a test to verify this:
https://github.com/xiaodongdu/kafka/blob/kafka-9960-kip-606/core/src/test/scala/unit/kafka/server/KafkaMetricsReporterTest.scala

Reply:
that's not what I meant. I meant for Kafka brokers we do not add any of those properties, so the line should be removed

Reply:
Fixed.